.github/scripts/generate_macos_ci_workflows.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python3
+
+import os
+from pathlib import Path
+
+import jinja2
+
+REPO_DIR = Path(__file__).absolute().parent.parent.parent
+GITHUB_DIR = REPO_DIR.joinpath(".github")
+TEMPLATES_DIR = GITHUB_DIR.joinpath("templates")
+
+
+def repo_path(filename):
+    return filename.replace(str(REPO_DIR), "").lstrip(os.path.sep)
+
+
+class MacOSWorkflow:
+    def __init__(self, build_environment: str, runner_type: str, on_pull_request: bool):
+        self.build_environment = build_environment
+
+        self.args = {
+            "build_environment": build_environment,
+            "runner_type": runner_type,
+            "on_pull_request": on_pull_request,
+        }
+
+    def generate_workflow_file(
+        self, workflow_template: jinja2.Template, jinja_env: jinja2.Environment
+    ) -> Path:
+        output_file_path = GITHUB_DIR.joinpath(
+            f"workflows/{self.build_environment}.yml"
+        )
+        with open(output_file_path, "w") as output_file:
+
+            output_file.writelines(
+                [
+                    "# @generated DO NOT EDIT MANUALLY\n",
+                    f"# Template is at:    {repo_path(workflow_template.filename)}\n",
+                    f"# Generation script: {repo_path(os.path.abspath(__file__))}\n",
+                ]
+            )
+            output_file.write(
+                workflow_template.render(**self.args)
+            )
+            output_file.write("\n")
+        return output_file_path
+
+
+WORKFLOWS = [
+    MacOSWorkflow(
+        build_environment="pytorch-macos-10.15",
```

Comment:
https://github.com/pytorch/pytorch/issues/57686#issuecomment-833672132
```suggestion
        build_environment="pytorch-macos-10.15-py3.8",
```